api/builder/Dockerfile

```diff
@@ -0,0 +1,38 @@
+FROM debian
```

Comment:
Being more explicit about the exact starting image definitely makes sense. I've selected `FROM golang:1.21.12-bookworm`.

With respect to doing a multi-stage build on top of distroless in order to shrink the resulting image, that makes sense for the lightnode use case, since the final image doesn't need the build environment. But for this PR, the image intentionally contains the build environment, since the the image is used to compile protobufs.

Reply:
I'm encountering difficulty trying to get this working on top of distroless. The core of the issue is that the tools required to build the protobuffs have a bunch of various dependencies, and it's notrivial to copy those all over to a clean image. Unlike the example in [smallest-golang-docker-image](https://klotzandrew.com/blog/smallest-golang-docker-image/), this is not a container where we are just trying to deploy a simple executable. We need our entire build toolchain, and that isn't packaged in a way that makes it simple to copy over.

The size of the dockerfile built on top of `golang:1.21.12-bookworm` is ~1gb. The size of a `distroless` image that just contains the go and protoc dependencies is ~0.33gb. Decently smaller, but not orders of magnitude smaller. Since this is something in the build pipeline and not something that we are going to need to ship to customers, IMO reducing the size of the container is not urgent.

Based on this information, would you be ok proceeding with this image as I currently have it?

Reply:
I think that given we are currently using this as a utility, with the committed protobufs remaining the standard of truth, I'm happy to not spend more time trying to fine-tune our choice of an image. @pschork, please let us know if you have additional thoughts.

Reply:
@pschork 
> Can we also update the root Makefile:protoc target to use this new build approach?

I plan to do this, but I am intentionally not doing it as a part of this PR. Changing the build process for everybody has the potential of being disruptive to CI pipelines and manual workflows, and I'd rather split high risk changes like that into their own PRs.